docker-compose.yml

```diff
@@ -24,6 +25,14 @@ services:
     volumes:
       - "react-web:/react-web"
     command: "bash -c ' rm -rf /react-web/* && cp -a . /react-web/'"
+  generate-static-rest:
+    build: .
+    image: flask_server
+    volumes:
+      - "./data:/covid19/data:ro"
+      - "static-rest:/covid19/rest"
+    command: "covid_web.generate_static_rest /covid19/data/ /covid19/rest/"
```

Comment:
What about generating those files in `download_all.zsh`?

Reply:
That's fine too

Reply:
I don't think that's a good place to do it, that script is just for downloading data. What you have here is data transformed into something else. For example you'd have to run `download_all.zsh` every time you change something in your code, which doesn't make sense conceptually.

The entry point is the right place to do it IMHO.

Reply:
`download_all.zsh` is not just downloading the data. It's also generating data and prediction protobuf files. If you modify something in `generate_predictions` you have to rerun script or post-processing command. It also make sense to put it here, as static REST files needs to be regenerated only when we have new data or updated predictions. If I'm not missing something, that's done only in `download_all.zsh`.

I suggest to split `download_all.zsh` into two files. One for downloading and second for post-processing (predictions + rest). `download_all.zsh` would also call postprocessing script.

Reply:
From the point of view of the graphs, the predictions are just data. In a hypothetical app with a database, you'd put country data into a Postgres DB together with the predictions. The predicitons would be calculated every night and put into the database.

However, you wouldn't put JSON files into the database (at least I don't see a reason for doing that).

Reply:
Yep, JSON in database does not make sense. Applied new entry point solution, since it's running fast enough, it can be executed every time we start the flask server.